### DISCUSSION TITLE
Jamulus doesn't start with crontab @reboot (Ubuntu 22.04)

### DISCUSSION THREAD

**Discussion by juanriccio** [jamulussoftware/jamulus#3141] [Technical Support & Help]
Hello, I have a dedicated machine for my Jamulus server. It runs Ubuntu LTS 22.
The server works fine, but my problem is that I can't have the server started each time the machine reboots.
I have added this to my personal (not root's) crontab, but Jamulus doesn't start. The same command works perfectly when run manually.

``
@reboot nohup /usr/bin/jamulus --server --numchannels 8 --log /home/juan/jamulus/jamulus.log --multithreading --startminimized --htmlstatus /home/juan/jamulus/jamulus-status.html --welcomemessage /home/juan/jamulus/jamulus-welcome.html 
``

I've tried with and without the nohup, but it doesn't make any difference.

Can anyone help please?

**Comment by chrisrimple** [comment]
Love this idea.  Every time I manually reboot my server, I then manually run:

`systemctl start jamulus`

But I'd love to have it automatically executed on every reboot, and even like the idea of force-rebooting the server nightly (via crontab or otherwise, if that's possible).

**Reply by pljones** [reply]
`systemctl enable jamulus` should tell systemd you really do want Jamulus started on reboot.  (I don't use the provided systemd files.)

**Reply by juanriccio** [reply]
Hello @pljones, and thanks for chiming in with your suggestion.
Unfortunately, it generates even more questions.

Your suggested command `systemctl enable jamulus` doesn't work on my system as it is. I guess it's because jamulus is not a registered service. How do I go about registering it? And once I do, will jamulus be run by root or by the plain user juan? I'd like it to keep it owned by juan, so the recording files are accessible without root credentials.

I'm not trying to wrestle you into a long winded explanation. A pointer to a page explaining the systemctl thing in a form suitable for non-sysadmins would likely be enough.

Thank you again!

**Reply by pljones** [reply]
https://jamulus.io/wiki/Running-a-Server#installation-and-configuration particularly the "Installation" section.  Like I say, I don't use the provided files, so you're better of reading the wiki if you're installing from the provided files.

**Reply by mcfnord** [reply]
If you install headless Jamulus and use it with **systemctl**, your path commands go in a configuration file that you will edit with this command:

`sudo systemctl edit --full jamulus-headless`

You'll put your command line into the **ExecStart** line.

it will run as user **jamulus**. You can change this by changing the User value to **juan**. However, these commands still won't work: **--log, --htmlstatus, --welcomemessage**. 

To make those work, set **ProtectHome** and maybe also **ProtectSystem** to **false**.

And I thought that would run at bootup, but it does not. !?!?!

**Reply by pljones** [reply]
The
```
[Install]
WantedBy=multi-user.target
```
in the systemd file should mean that when you run the `systemctl enable` command, you get a link created in `/etc/systemd/system/multi-user.target.wants` with the name of the `.service` file.

Check that file exists after running the `enable` before rebooting.  Try manually starting after the enable, too, and check with `journalctl -u <service> --since <ten minutes ago> --follow` that the service is running.  After rebooting, check the `systemctl status` output and the `journalctl` output.
